backend/src/apiserver/list/list.go

```diff
@@ -226,21 +234,23 @@ func (o *Options) AddFilterToSelect(sqlBuilder sq.SelectBuilder) sq.SelectBuilde
 }
 
 // FilterOnResourceReference filters the given resource's table by rows from the ResourceReferences
-// table that match an optional given filter, and returns the rebuilt SelectBuilder
+// table that match an optional given filter, and returns the rebuilt SelectBuilder.
 func FilterOnResourceReference(tableName string, columns []string, resourceType model.ResourceType,
-	selectCount bool, filterContext *common.FilterContext) (sq.SelectBuilder, error) {
+	selectCount bool, filterContext *model.FilterContext,
+) (sq.SelectBuilder, error) {
 	selectBuilder := sq.Select(columns...)
 	if selectCount {
 		selectBuilder = sq.Select("count(*)")
 	}
 	selectBuilder = selectBuilder.From(tableName)
-	if filterContext.ReferenceKey != nil {
+	if filterContext.ReferenceKey != nil && filterContext.ReferenceKey.ID != "" {
```

Comment:
This is being called in `storage` package. The change enables retrieval of resources in single-user mode, where we need to avoid filtering on the namespace value.

In `v1beta1`, namespace could be missing in resource references and we didn't need to check for the actual value of the `ReferenceKey.ID`. In `v2beta1`, resource are namespaced (sometimes required field). This gets assigned to an empty string during conversion to `model` types. So now, we need to check the value of the key.

`filterContext.ReferenceKey.ID` is the same as `filterContext.ID`, but it shows more information about the structure. This can be removed.

I will add more comments to the changes. Thank you for the suggestion.